ui/src/pages/custody/Assets.tsx

```diff
@@ -12,62 +10,37 @@ import { ServicePageProps, createDropdownProp } from '../common';
 import { usePartyName } from '../../config';
 import { useLedger, useParty } from '@daml/react';
 import { useDisplayErrorMessage } from '../../context/MessagesContext';
-import TitleWithActions from '../../components/Common/TitleWithActions';
-import { NewAccount } from './New';
-
-const Assets: React.FC<ServicePageProps<Service>> = ({ services }: ServicePageProps<Service>) => {
-  const { contracts: accounts, loading: accountsLoading } = useStreamQueries(AssetSettlementRule);
-  const { contracts: allocatedAccounts, loading: allocatedAccountsLoading } =
-    useStreamQueries(AllocationAccountRule);
-  const { contracts: assets } = useStreamQueries(AssetDescription);
-  const displayErrorMessage = useDisplayErrorMessage();
-
-  const [creditAsset, setCreditAsset] = useState<string>('');
-  const [creditQuantity, setCreditQuantity] = useState<string>('');
-  const [selectedAccount, setSelectedAccount] = useState<string>('');
 
+const Assets: React.FC<ServicePageProps<CustodyService>> = ({
+  services,
+}: ServicePageProps<CustodyService>) => {
   const party = useParty();
   const ledger = useLedger();
-
   const { getName } = usePartyName(party);
 
-  const allAccounts = useMemo(
-    () =>
-      accounts
-        .map(a => {
-          return { account: a.payload.account, contractId: a.contractId.replace('#', '_') };
-        })
-        .concat(
-          allocatedAccounts.map(a => {
-            return { account: a.payload.account, contractId: a.contractId.replace('#', '_') };
-          })
-        ),
-    [accounts, allocatedAccounts]
+  const { contracts: allAssets } = useStreamQueries(AssetDescription);
+  const clientServices = services.filter(s => s.payload.customer === party);
+  const assets = allAssets.filter(a =>
+    clientServices.map(c => c.payload.provider).includes(a.payload.registrar)
   );
+  const displayErrorMessage = useDisplayErrorMessage();
 
-  if (accountsLoading || allocatedAccountsLoading) {
-    return <div>Loading...</div>;
-  }
-
-  const clientServices = services.filter(s => s.payload.customer === party);
+  const [creditAsset, setCreditAsset] = useState<string>('');
+  const [creditQuantity, setCreditQuantity] = useState<string>('');
 
-  const onRequestCredit = async () => {
+  const onRequestDeposit = async () => {
     const asset = assets.find(i => i.payload.description === creditAsset);
-    const targetAccount = allAccounts.find(a => a.contractId === selectedAccount);
-    if (!asset || !targetAccount) return;
-    const service = clientServices.find(
-      s => s.payload.provider === targetAccount?.account.provider
-    );
 
+    if (!asset) return;
```

Comment:
does this return warrant an error message?

Reply:
We could refactor the code to avoid having to do a `find` on the list of available assets so we don't need this check. I'd prefer to do that as part of a separate PR if that's ok?